common/server_shm.hpp

```diff
@@ -28,6 +28,12 @@
 #include <boost/interprocess/managed_shared_memory.hpp>
 #include <boost/interprocess/containers/vector.hpp>
 
+#ifdef _WIN32
+#    include "SC_Filesystem.hpp"
+#    define BOOST_INTERPROCESS_SHARED_DIR_FUNC                                                                         \
+        void get_shared_dir(std::string& shared_dir) { shared_dir = SC_Filesystem::defaultResourceDirectory + "/shm" }
```

Comment:
according to boost docs, what you should be doing is

`#define BOOST_INTERPROCESS_SHARED_DIR_FUNC`

prior to including interprocess headers, and then implementing the function

```c++
namespace boost {
    namespace interprocess {
        namespace ipcdetail {
            void get_shared_dir(std::string &shared_dir);
        }
    }
}
```

somewhere (i would do it directly above `namespace detail_server_shm` as you have done here). you would also make that function `inline` since this is a header.

i think it would be safest to set this preprocessor definition in the build system so that it is defined in every file on Windows. there is only one other file in the project that uses B.Interprocess, and it doesn't use these headers, but that may change in the future.

Reply:
> i think it would be safest to set this preprocessor definition in the build system

I don't think I understand. Would the actual function body go into cmake?

Reply:
sorry, what I'm saying is that this feature is not intended to be used like:

```cpp
#define BOOST_INTERPROCESS_SHARED_DIR_FUNC myFunction() {}
```

but rather

```cpp
// in header
#define BOOST_INTERPROCESS_SHARED_DIR_FUNC
#include <boost/interprocess/_____.hpp>

// in cpp
namespace boost { namespace interprocess { namespace ipcdetail {
void get_shared_dir(std::string &shared_dir) { /* impl */ }
}}}
```

and that setting `BOOST_INTERPROCESS_SHARED_DIR_FUNC` in cmake is slightly more robust than doing it in the header file.